docs/source/developers/guide/step_by_step/building.rst

```diff
@@ -0,0 +1,37 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _build-arrow:
+
+*************************
+Building Arrow package 🏋🏿‍♀️
```

Comment:
We already have (or should have) such documentation in per-language docs, so I don't think duplicating it is useful (and it will be harder to maintain as well).

Reply:
What I meant to add here were short snippets of information explaining what are we doing when we want to build Arrow (why building C++ separately, what is cmake, ...)

The details of the build in the existing docs will be linked from here.